### PR TITLE
fix(#302): Use HASS-Standard Units on Furnace Entities

### LINKED ISSUE #302: Units Not Matching HASS Expectations for Some HVAC Sensors
Discord user `@fireheadman` reported the following in his HASS logs:
```
2023-12-28 14:49:01.075 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.econet_hvac_furnace_odu_pressure_suction (<class 'homeassistant.components.esphome.sensor.EsphomeSensor'>) is using native unit of measurement 'PSI' which is not a valid unit for the device class ('pressure') it is using; expected one of ['hPa', 'Pa', 'bar', 'cbar', 'mmHg', 'inHg', 'mbar', 'psi', 'kPa']; Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+esphome%22
2023-12-28 14:51:14.100 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.econet_hvac_furnace_flame_sensor (<class 'homeassistant.components.esphome.sensor.EsphomeSensor'>) is using native unit of measurement 'uA' which is not a valid unit for the device class ('current') it is using; expected one of ['A', 'mA']; Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+esphome%22
2023-12-28 14:51:14.122 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.econet_hvac_furnace_furnace_high_heat_btu_output (<class 'homeassistant.components.esphome.sensor.EsphomeSensor'>) is using native unit of measurement 'btu/h' which is not a valid unit for the device class ('power') it is using; expected one of ['kW', 'W']; Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+esphome%22
2023-12-28 14:51:14.139 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.econet_hvac_furnace_furnace_low_heat_btu_output (<class 'homeassistant.components.esphome.sensor.EsphomeSensor'>) is using native unit of measurement 'btu/h' which is not a valid unit for the device class ('power') it is using; expected one of ['kW', 'W']; Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+esphome%22
2023-12-28 14:51:14.461 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.econet_hvac_furnace_instant_btu_output (<class 'homeassistant.components.esphome.sensor.EsphomeSensor'>) is using native unit of measurement 'kbtu/h' which is not a valid unit for the device class ('power') it is using; expected one of ['kW', 'W']; Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+esphome%22
2023-12-28 14:51:14.480 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.econet_hvac_furnace_instant_btu_usage (<class 'homeassistant.components.esphome.sensor.EsphomeSensor'>) is using native unit of measurement 'kbtu/h' which is not a valid unit for the device class ('power') it is using; expected one of ['kW', 'W']; Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+esphome%22
2023-12-28 14:51:14.537 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.econet_hvac_furnace_indoor_kw (<class 'homeassistant.components.esphome.sensor.EsphomeSensor'>) is using state class 'measurement' which is impossible considering device class ('energy') it is using; expected None or one of 'total_increasing', 'total'; Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+esphome%22
2023-12-28 14:51:14.537 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.econet_hvac_furnace_indoor_kw (<class 'homeassistant.components.esphome.sensor.EsphomeSensor'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('energy') it is using; expected one of ['kWh', 'MJ', 'MWh', 'GJ', 'Wh']; Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+esphome%22
2023-12-28 14:51:14.772 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.econet_hvac_furnace_odu_pressure_suction (<class 'homeassistant.components.esphome.sensor.EsphomeSensor'>) is using native unit of measurement 'PSI' which is not a valid unit for the device class ('pressure') it is using; expected one of ['hPa', 'Pa', 'bar', 'cbar', 'mmHg', 'inHg', 'mbar', 'psi', 'kPa']; Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+esphome%22
```

Seems like we may need to update some of our configurations with unit conversions to make sure we are using units listed at https://www.home-assistant.io/integrations/sensor/

### PR DESCRIPTION
**Pull request by barndawgie**
Fix #302 but updating/converting some units.

Setting as draft for now pending discussion of a few points in #302.

### CODE REVIEW

File: econet_hvac_furnace.yaml [hunk 1]
```diff
@@ -108,9 +108,10 @@ sensor:
     device_class: "current"
     entity_category: "diagnostic"
     accuracy_decimals: 1
-    unit_of_measurement: "uA"
+    unit_of_measurement: "mA"
     filters:
       - delta: 0.09
+      - multiply: .001 # Convert from uA to mA
```
Comment (tronikos) [inline econet_hvac_furnace.yaml:114]: Since this is a template sensor this could go where this is calculated. Does the delta need to be adjusted?

Comment (barndawgie) [reply econet_hvac_furnace.yaml:114]: Filters are run in order, so the Delta wouldn't need to be adjusted if implemented as shown. If I move it the Delta would need to be adjusted.

This is where it's set: https://github.com/esphome-econet/esphome-econet/blob/318f261d7a87711a2e46be3cbcffb08f90487c5b/econet_hvac_furnace.yaml#L32

@stockmopar what's the reason we divide by 10 there? Does it provide the current in 1/10 uA units by default?

Comment (stockmopar) [reply econet_hvac_furnace.yaml:114]: Yep it does provide current to tenths of uA.

**Review by stockmopar — APPROVED** [review]
Looks good.

**Pull request merged**